src/network/https.js

```diff
@@ -165,6 +240,15 @@ class HttpsWorker {
           // The refresh token is not valid.
           throw new Error(`Unable to refresh expired token: ${error}`);
         }
+        try {
+          // and call the handler for this event if we have one
+          if (typeof (this.events.refresh) === 'function') {
+            await this.events.refresh(this.tokens);
```

Comment:
you don't need to call the handler here since `this.refreshAuthentication()` is called above and it then calls the handler. I just confirmed and the refresh handler is being called twice on implicit refresh.